chromium_src/content/browser/web_contents/web_contents_impl.cc

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "content/browser/web_contents/web_contents_impl.h"
+
+namespace content {
+
+bool WebContentsImpl::ShouldDoLearning() {
+  if (!ShouldDoLearning_ChromiumImpl()) {
+    return false;
+  }
+  return GetContentClient()
+      ->browser()
+      ->IsWindowsRecallEnabled(GetBrowserContext())
+      .value_or(true);
+}
+
+void WebContentsImpl::GetShouldDoLearningForTesting(bool& should) {
```

Comment:
why are you using an out param here? https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs `Prefer using return values over output parameters` but also in particular this seems like a very odd usage of out params